ncimonitor/UsageDataset.py

```diff
@@ -314,3 +314,39 @@ def getsystemstorage(self, systemname, storagepoint, year, quarter):
         if q is None:
             return (None,None)
         return float(q['grant']),float(q['igrant'])
+
+
+    def top_usage(self, storagepoint, count=10, date=None, measure='size'):
+        """
+        Return the top ``count`` users according to ``measure`` (either 'size'
+        or 'inodes') on ``storagepoint`` at ``date``
+
+        Due to the NCI scanning process ``date`` should be at least one day in
+        the past
+
+        Returns columns (measure, username, fullname)
+        """
+
+        if storagepoint not in ['short', 'gdata']:
+            raise Exception(f"Unexpected storagepoint '{storagepoint}'")
+
+        if measure not in ['size', 'inodes']:
+            raise Exception(f"Unexpected measure '{measure}'")
+
+        if date is None:
+            # Default to one day in the past
+            date = datetime.date.today() - datetime.timedelta(days=1)
+
+        q = self.db.query(f"""
```

Comment:
I was sort of hoping to converge on minimising SQL queries and use the pandas dataframes as a unit of interchange. For example, the same functionality can be done by calling 

```
(Pdf) dp = db.getstorage(year, quarter, storagept=storagept, datafield=datafield)
(Pdb) p dp.ix[-1].sort_values().tail(5)
Name
G**** L**** (gl****)      6.536940e+12
N**** C**** (nc****)      7.505455e+12
A**** H**** (am****)      8.724162e+12
P**** S**** (pa****)      2.006201e+13
A**** K**** (ae****)      2.430116e+13
```

Reply:
The rationale is to make it easier for others to add functionality without SQL knowledge, as they can follow examples like above.